books/serializers.py

```diff
@@ -0,0 +1,77 @@
+from rest_framework import serializers
+
+from books.models.books import Book, Author, Publisher
+from books.models.country import Country
+
+
+class AuthorSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=250)
+
+    class Meta:
+        model = Author
+        fields = ('id', 'name')
+
+
+class PublisherSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=250)
+
+    class Meta:
+        model = Publisher
+        fields = ('id', 'name')
+
+
+class CountrySerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=250)
+
+    class Meta:
+        model = Country
+        fields = ('id', 'name')
+
+
+class BookSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    publisher = PublisherSerializer(required=True)
+    country = CountrySerializer(required=True)
+    authors = AuthorSerializer(required=True, many=True)
+
+    class Meta:
+        model = Book
+        fields = ("id", "name", "isbn", "authors", "number_of_pages", "publisher", "country", "release_date")
+
+    def create(self, validated_data):
+        authors = validated_data.pop('authors')
```

Comment:
@xhahid43eb This is not proper use of restful API, book serializer should be only responsible for creating book object with given data. It is the responsibility of API user to provide correct data. So we shouldn't be pulling country, author, publisher from DB and creating Book instance manually with that data. IMO what book api should be passed correctly with these values (authors id(s), country id, and publisher id) in request data and serializer should create book entry using that data. The approach you used is bad. Lets suppose there are 15 different models associated with book other than author,publisher and country. Then this approach will definitely cost 15 different get requests to DB when we are going to create a single book record. With increasing complexity this design will make your app perfom slow. Please update these serializers accordingly.